docs/unit-test-infrastructure.md

```diff
@@ -0,0 +1,227 @@
+# Unit Test Infrastructure 
```

Comment:
This is now duplicate information that will need to be maintained and kept in sync in two places. So it will get out of sync with the very next change.

This problem needs to be solved.

I'm honestly not sure if we really need this document. You'll master the notations after your 1st sprint and then basically never come back to it again.

Reply:
I think the scope of this document was to be available to anyone who might be interested in our Unit Test Infrastructure, either to be able to understand unit tests or to write some on their own.

That being said, I agree with the certainty that it will get out-of-sync with the comment section that it references.

Maybe we could keep it as an internal document, just for reference, and archive/delete this PR?

Reply:
External PRs are rare. And the testing framework can be used intuitively just by looking at other examples.

Even internal document will get out of sync. I'd just close it. The content will be thought to every newjoiner as part of onboarding.